modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -370,11 +370,14 @@ impl ServerModule for Wallet {
                 let new_consensus_block_count = self.consensus_block_count(dbtx).await;
 
                 // only sync from the first non-default consensus block count
-                if new_consensus_block_count > old_consensus_block_count {
+                if old_consensus_block_count.is_some()
+                    && new_consensus_block_count.is_some()
+                    && new_consensus_block_count.unwrap() > old_consensus_block_count.unwrap()
+                {
                     self.sync_up_to_consensus_height(
                         dbtx,
-                        old_consensus_block_count,
-                        new_consensus_block_count,
+                        old_consensus_block_count.unwrap(),
+                        new_consensus_block_count.unwrap(),
```

Comment:
I'm not a big fan of the unwrapping here. A nested if-let+if or a match statement would be nicer:

```rust
match (old_consensus_block_count, new_consensus_block_count) {
    (Some(old), Some(new)) if new > old => {
         self.sync_up_to_consensus_height(dbtx, old, new).await;
    }
    _ => {}
}
```

Reply:
Ascending order better, IMO. `old < new` 


```rust
    if let (Some(old), Some(new))  = (old_consensus_block_count, new_consensus_block_count){
        if old < new {
            println!("yes");
        }
    }
```

Unfortunately `&& old < new` is not yet stable.

Reply:
Let's agree to disagree :smile: